net: implement load_balancing_algorithm::fixed

cyx1231st opened the pull request:
Add a new load_balancing_algorithm::fixed to force server_socket to accept connected_socket on a fixed CPU.

rzarzynski commented on src/net/posix-stack.cc:
```diff
@@ -427,8 +427,17 @@ posix_server_socket_impl::accept() {
-        auto cth = _lba == server_socket::load_balancing_algorithm::connection_distribution ?
-                _conntrack.get_handle() : _conntrack.get_handle(ntoh(sa.as_posix_sockaddr_in().sin_port) % smp::count);
+        auto cth = [this, &sa] {
+            switch(_lba) {
```
:+1: for improving the code readability.

tchaikov replied:
i'd suggest add a space after `switch`.

rzarzynski reviewed (commented):
I would advice to split the PR into two commits:
   * the clean-up,
   * introduction of `load_balancing_algorithm::fixed`

and send the former one without waiting on the latter.

cyx1231st committed:
net: posix: Introduce load_balancing_algorithm::fixed

Add a new policy load_balancing_algorithm::fixed to force server_socket to
accept connected_socket on a fixed shard.

Signed-off-by: Yingxin Cheng <yingxin.cheng@intel.com>

cyx1231st commented:
Updated related documentation. I didn't split the commit because I think it's acceptable to use conditional ternary operator when there are only two cases, except that the line is too long and could be wrapped.

tchaikov commented:
closing, as it's been upstreamed.
